outputs/spyderbat.go

```diff
@@ -0,0 +1,216 @@
+package outputs
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/falcosecurity/falcosidekick/types"
+	"github.com/google/uuid"
+)
+
+func isSourcePresent(config *types.Configuration) (bool, error) {
+
+	client := &http.Client{}
+
+	req, err := http.NewRequest("GET", config.Spyderbat.APIUrl+"/api/v1/org/"+config.Spyderbat.OrgUID+"/source/", new(bytes.Buffer))
+	if err != nil {
+		return false, err
+	}
+	req.Header.Add("Authorization", "Bearer "+config.Spyderbat.APIKey)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return false, errors.New("HTTP error: " + resp.Status)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false, err
+	}
+	var sources []map[string]interface{}
+	if err := json.Unmarshal(body, &sources); err != nil {
+		return false, err
+	}
+	uid := "falcosidekick_" + config.Spyderbat.OrgUID
+	for _, source := range sources {
+		if id, ok := source["uid"]; ok && id.(string) == uid {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+type SourceBody struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	UID         string `json:"uid"`
+}
+
+func makeSource(config *types.Configuration) error {
+
+	data := SourceBody{
+		Name:        config.Spyderbat.Source,
+		Description: config.Spyderbat.SourceDescription,
+		UID:         "falcosidekick_" + config.Spyderbat.OrgUID,
+	}
+	body := new(bytes.Buffer)
+	if err := json.NewEncoder(body).Encode(data); err != nil {
+		return err
+	}
+
+	client := &http.Client{}
+
+	req, err := http.NewRequest("POST", config.Spyderbat.APIUrl+"/api/v1/org/"+config.Spyderbat.OrgUID+"/source/", body)
```

Comment:
There's already method to POST, can you see to use it instead please https://github.com/falcosecurity/falcosidekick/blob/master/outputs/client.go#L136

Reply:
I could do that, however it is a function on a client instance (not http), and at that point, the client hasn't been created yet - the client POST method also seems to be for the purpose of sending data to the output, whereas this post is to create a source to receive the output, so using it for this would require a bit of workaround of that method.
Is the point of using that method to ensure the ContentType and UserAgent headers are present, and that the TLS and anything else in the config is reflected in the request? If so, I feel like it would be preferable to extract that into its own function, so that it can be used on my post, the original post method, and then also on the GET request I make

Reply:
You're right, my mistake, all seems good to me. The blocking PR has been merged and will be released with the embedded in Falco 0.33 (supposed to be out today).